File: 19-08-2025_Number_of_Zero-Filled_Subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        count = 0
        zero_length = 0
    
        for num in nums:
            if num == 0:
                zero_length += 1
                count += zero_length
            else:
                zero_length = 0
    
        return count
    

File: 19-08-2025_Number_of_Zero-Filled_Subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long count = 0;
        long long zeroLength = 0;

        for (int num : nums) {
            if (num == 0) {
                zeroLength++;
                count += zeroLength;
            } else {
                zeroLength = 0;
            }
        }

        return count;
    }
};
